android/src/play/java/com/dooboolab/RNIap/RNIapModule.kt

```diff
@@ -81,30 +80,27 @@ class RNIapModule(
                 TAG,
                 "Already initialized, you should only call initConnection() once when your app starts"
             )
-            promise.resolve(true)
+            promise.safeResolve(true)
             return
         }
         if (googleApiAvailability.isGooglePlayServicesAvailable(reactContext)
             != ConnectionResult.SUCCESS
         ) {
             Log.i(TAG, "Google Play Services are not available on this device")
-            promise.reject(DoobooUtils.E_NOT_PREPARED, "Google Play Services are not available on this device")
+            promise.safeReject(DoobooUtils.E_NOT_PREPARED, "Google Play Services are not available on this device")
```

Comment:
Could you pin me on the link where I can learn more about `safeReject` and `safeResolve`?

I want to be more specific about when to use these 🤔

Reply:
@andresesfm Looks like the method does not exist anymore in [react-native repo, Promise.java](https://github.com/facebook/react-native/blob/main/ReactAndroid/src/main/java/com/facebook/react/bridge/Promise.java).

Reply:
It's a kotlin extension method I created in PromiseUtils. It's more convenient so se don't have the try-catch block for promises everywhere

Reply:
Oh. It looks cool!